app/models/user.rb

```diff
@@ -27,7 +27,9 @@ class User < ApplicationRecord
   after_save :send_welcome_email, if: :needs_welcome_email?
   before_destroy :clear_swap
 
-  validates :email, uniqueness: { case_sensitive: false }, allow_nil: true
+  # Additional validation for emails, :validatable has already added basic validation
+  validate :email_uniqueness, on: :create
```

Comment:
We only validate on create, which should prevent some of the problems we're seeing. `:validatable` only validates on change:

```
        base.class_eval do
          validates_presence_of   :email, if: :email_required?
          if Devise.activerecord51?
            validates_uniqueness_of :email, allow_blank: true, case_sensitive: true, if: :will_save_change_to_email?
            validates_format_of     :email, with: email_regexp, allow_blank: true, if: :will_save_change_to_email?
          else
            validates_uniqueness_of :email, allow_blank: true, if: :email_changed?
            validates_format_of     :email, with: email_regexp, allow_blank: true, if: :email_changed?
          end

          validates_presence_of     :password, if: :password_required?
          validates_confirmation_of :password, if: :password_required?
          validates_length_of       :password, within: password_length, allow_blank: true
        end
      end
```